test/test_interpolations.jl

```diff
@@ -252,6 +249,251 @@ using Ferrite: reference_shape_value, reference_shape_gradient
         end
     end
 
+    reference_cell(::Type{RefTriangle}) = Triangle((1, 2, 3))
+    reference_cell(::Type{RefQuadrilateral}) = Quadrilateral((1, 2, 3, 4))
+    reference_cell(::Type{RefTetrahedron}) = Tetrahedron((1, 2, 3, 4))
+    reference_cell(::Type{RefHexahedron}) = Hexahedron((ntuple(identity, 8)))
+
+    function line_integral(qr::QuadratureRule{RefLine}, ip, shape_nr, x0, Δx, L, v, f)
+        val = 0.0
+        for (ξ1d, w) in zip(Ferrite.getpoints(qr), Ferrite.getweights(qr))
+            ξ = x0 + (ξ1d[1] / 2) * Δx  # ::Vec
+            s = (ξ1d[1] + 1) / 2        # ∈ [0, 1]
+            Nξ = reference_shape_value(ip, ξ, shape_nr)
+            dΩ = (w * L / 2)
+            val += (Nξ ⋅ v) * f(s) * dΩ
+        end
+        return val
+    end
+
+    # Required properties of shape value Nⱼ of an edge-elements (Hcurl) on an edge with direction v, length L, and dofs ∈ 𝔇
+    # 1) Unit property: ∫(Nⱼ ⋅ v f(s) dS) = 1/length(𝔇) ∀ ∈ 𝔇
+    #    Must hold for
+    #    length(𝔇) ≥ 1: f(s) = 1
+    #    length(𝔇) = 2: f(s) = 1 - s or f(s) = s for 1st and 2nd dof, respectively.
+    #    Additionally, should be zero for
+    #    length(𝔇) = 2: f(s) = s or f(s) = 1 - s for 1st and 2nd dof, respectively.
+    #    s is the path parameter ∈[0,1] along the positive direction of the path.
+    # 2) Zero along other edges: Nⱼ ⋅ v = 0 if j∉𝔇
```

Comment:
Associated test definitions for [comment above](https://github.com/Ferrite-FEM/Ferrite.jl/pull/798/files#r1854906581) for Hcurl